cmd/goal/node.go

```diff
@@ -374,3 +394,97 @@ var waitCmd = &cobra.Command{
 		}
 	},
 }
+
+func isValidIP(userInput string) bool {
+	host, port, err := net.SplitHostPort(userInput)
+	if err != nil {
+		return false
+	}
+	if port == "" {
+		return false
+	}
+	if host == "" {
+		return false
+	}
+	return isValidIP(host)
+}
+
+var createCmd = &cobra.Command{
+	Use:   "create",
+	Short: "create a node at the desired data directory for the desired network",
+	Long:  "create a node at the desired data directory for the desired network",
+	Args:  validateNoPosArgsFn,
+	Run: func(cmd *cobra.Command, _ []string) {
+
+		// validate network input
+		validNetworks := map[string]bool{"mainnet": true, "testnet": true, "devnet": true}
+		if !validNetworks[newNodeNetwork] {
+			reportErrorf(errorNodeCreation, "passed network name invalid")
+		}
+
+		// locate genesis block
+		exePath, err := util.ExeDir()
+		if err != nil {
+			reportErrorln(errorNodeCreation, err)
+		}
+		firstChoicePath := filepath.Join(exePath, "genesisfiles", newNodeNetwork, "genesis.json")
+		secondChoicePath := filepath.Join("var", "lib", "algorand", "genesis", newNodeNetwork, "genesis.json")
+		thirdChoicePath := filepath.Join(exePath, "genesisfiles", "genesis", newNodeNetwork, "genesis.json")
+		paths := []string{firstChoicePath, secondChoicePath, thirdChoicePath}
+		correctPath := ""
+		for _, pathCandidate := range paths {
+			if util.FileExists(pathCandidate) {
+				correctPath = pathCandidate
+				break
+			}
+		}
+		if correctPath == "" {
+			reportErrorf("Could not find genesis.json file. Paths checked: %v", strings.Join(paths, ","))
+		}
+
+		// verify destination does not exist, and attempt to create destination folder
+		if util.FileExists(newNodeDestination) {
+			reportErrorf(errorNodeCreation, "destination folder already exists")
+		}
+		destPath := filepath.Join(newNodeDestination, "genesis.json")
+		err = os.MkdirAll(newNodeDestination, 0666)
+		if err != nil {
+			reportErrorf(errorNodeCreation, "could not create destination folder")
+		}
+
+		// copy genesis block to destination
+		genesisInput, err := ioutil.ReadFile(correctPath)
+		if err != nil {
+			reportErrorf(errorNodeCreation, err)
+		}
+		err = ioutil.WriteFile(destPath, genesisInput, 0666)
+		if err != nil {
+			reportErrorf(errorNodeCreation, err)
+		}
+
+		// build config and save to destination
+		localConfig := config.GetDefaultLocal()
+		localConfig.Archival = newNodeArchival || newNodeRelay != "" || newNodeIndexer
+		localConfig.IsIndexerActive = newNodeIndexer
+		localConfig.RunHosted = runUnderHost
+		if newNodeRelay != "" {
+			if isValidIP(newNodeRelay) {
```

Comment:
Validate this before creating destination directory so you can error out without any side-effects